R/metadata.R

```diff
@@ -163,6 +163,66 @@ get_calibration_plot_settings <- function(filter_types) {
   )
 }
 
+get_comparison_plot_settings_control <- function(filter_types) {
+  list(
+    comparison = get_comparison_plot_settings(filter_types)
+  )
+}
+
+get_comparison_plot_settings <- function(filter_types) {
+  x_axis_filters <- c("period", "sex", "age")
+  default_filter_ids <- c(c("indicator", "area", "source"),
+                          x_axis_filters)
+  default_filters <- lapply(default_filter_ids, get_filter_from_id)
+  all_filters <- c(default_filters, list(get_filter_from_id("detail")))
+
+  area_x_axis_effect <- list(
+    id = scalar("area"),
+    label = scalar(get_label_for_id("area")),
+    effect = list(
+      setMultiple = "area",
+      setFilters = all_filters
+    )
+  )
+  ## TODO: In current plot when you change indicator, it updates
+  ## the filters. We could support this same behaviour by making
+  ## indicator a plot control which updates the filter values
+  ## including a hidden "indicator" filter which would be the value
+  ## actually used for filtering the data. But let's check what we
+  ## actually want to do. Would have to set the x-axis too, which I don't
+  ## think we can support yet.
```

Comment:
i would say that this definitely sounds a little complicated, it would be nice to keep the complexity low in the effects because while it is powerful, i can see it going out of control very quickly.

my recommendation would be to change how the users have to interact with the comparison plot slightly and instead give them preset plot control with options like "Prevalence by age", "ART number by sex" and so on, so you get to preserve the pairings of the indicator with x axis and any other filter values you want to change while still being consistent with other plots

Reply:
Thanks, I'm going to make a note of this and let's discuss when we review with Jeff and Rachel on Thursday